pkg/shp/suggestion/suggest.go

```diff
@@ -0,0 +1,88 @@
+package suggestion
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/texttheater/golang-levenshtein/levenshtein"
+)
+
+// SubcommandsRequiredWithSuggestions will ensure we have a subcommand provided by the user and augments it with
+// suggestion for commands, alias and help on root command.
+func SubcommandsRequiredWithSuggestions(cmd *cobra.Command, args []string) error {
+	requireMsg := "unknown command %q for %q"
+	typedName := ""
+	// This will be triggered if cobra didn't find any subcommands.
+	// Find some suggestions.
+	var suggestions []string
+
+	if len(args) != 0 && !cmd.DisableSuggestions {
+		typedName += args[0]
+		if cmd.SuggestionsMinimumDistance <= 0 {
+			cmd.SuggestionsMinimumDistance = 2
+		}
+		// subcommand suggestions
+		suggestions = cmd.SuggestionsFor(args[0])
+
+		// subcommand alias suggestions (with distance, not exact)
+		for _, c := range cmd.Commands() {
+			if c.IsAvailableCommand() {
```

Comment:
nit: I would rather `if !c.IsAvailableCommand() { continue }`, so we avoid indentation levels, and make sure the conditions that would interrupt the loop processing are high up. However, a nit comment :-)

Reply:
addressed